rm-datasource/src/main/java/com/alibaba/fescar/rm/datasource/exec/SelectForUpdateExecutor.java

```diff
@@ -87,7 +87,9 @@ public Object doExecute(Object... args) throws Throwable {
                     }
 
                     TableRecords selectPKRows = TableRecords.buildRecords(getTableMeta(), rsPK);
-                    statementProxy.getConnectionProxy().checkLock(selectPKRows);
+                    if(!selectPKRows.getRows().isEmpty()) {
```

Comment:
I would prefer to put such check inside com.alibaba.fescar.rm.datasource.ConnectionProxy#checkLock

BTW, it's better to fire an issue before PR. 👍

Reply:
I see, thx!